clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -642,37 +645,35 @@ public long read(ByteBuffer[] dsts, int offset, int length) throws IOException {
     */
     @Override
     public int write(ByteBuffer src) throws IOException {
-        int written = 0;
         if (state == State.CLOSING)
             throw closingException();
         if (state != State.READY)
-            return written;
+            return 0;
```

Comment:
Also, while you are at it, it might be useful to change the comment for the return value to "The number of bytes read from src". Otherwise, it's bit confusing since the method is write().